lib/task.js

```diff
@@ -77,17 +78,8 @@ Create a list of cached volumes that will be mounted within the docker container
 @param {object} volumes to mount in the container
  */
 async function buildVolumeBindings(taskVolumeBindings, volumeCache, taskScopes) {
-  let neededScopes = [];
-
-  for (let volumeName in taskVolumeBindings) {
-    neededScopes.push('docker-worker:cache:' + volumeName);
-
-    if (!scopeMatch(taskScopes, neededScopes)) {
-      throw new Error(
-        'Insufficient scopes to attach "' + volumeName + '" as a cached ' +
-        'volume.  Try adding ' + neededScopes + ' to the .scopes array.'
-      );
-    }
+  if (!hasPrefixedScopes('docker-worker:cache:', taskVolumeBindings, taskScopes)) {
+    throw new Error('Insufficient scopes to attach cache volumes.');
```

Comment:
Agreed.  It wasn't quite "nice" since it didn't give the right set of scopes, though.  How about adding something like "the task must have scope `docker-worker:cache:<cache-name>` for each cache in `payload.caches`."?  Given that information, it's trivial for a user to figure out what scopes they need.